cordova-plugin-outline/apple/vpn/PacketTunnelProvider.m

```diff
@@ -117,55 +114,53 @@ - (void)startTunnelWithOptions:(NSDictionary *)options
   // culprit and can explicitly disconnect.
-  [self.shadowsocks
-      startWithConnectivityChecks:!isOnDemand
-                       completion:^(ErrorCode errorCode) {
-                         ErrorCode clientErrorCode =
-                             (errorCode == noError || errorCode == udpRelayNotEnabled) ? noError
-                                                                                       : errorCode;
-                         if (clientErrorCode == noError) {
-                           [self connectTunnel:[self getTunnelNetworkSettings]
-                                    completion:^(NSError *error) {
-                                      if (!error) {
-                                        BOOL isUdpSupported =
-                                            isOnDemand ? self.connectionStore.isUdpSupported
-                                                       : errorCode == noError;
-
-                                        if (![self startTun2Socks:isUdpSupported]) {
-                                          [self execAppCallbackForAction:kActionStart
-                                                               errorCode:vpnStartFailure];
-                                          return completionHandler([NSError
-                                              errorWithDomain:NEVPNErrorDomain
-                                                         code:NEVPNErrorConnectionFailed
-                                                     userInfo:nil]);
-                                        }
-                                        [self execAppCallbackForAction:kActionStart
-                                                             errorCode:noError];
-                                        [self listenForNetworkChanges];
-                                        [self.connectionStore save:connection];
-                                        self.connectionStore.isUdpSupported = isUdpSupported;
-                                        self.connectionStore.status = ConnectionStatusConnected;
-
-                                      } else {
-                                        [self execAppCallbackForAction:kActionStart
-                                                             errorCode:vpnPermissionNotGranted];
-                                      }
-                                      completionHandler(error);
-                                    }];
-                         } else {
-                           [self execAppCallbackForAction:kActionStart errorCode:clientErrorCode];
-                           completionHandler([NSError errorWithDomain:NEVPNErrorDomain
-                                                                 code:NEVPNErrorConnectionFailed
-                                                             userInfo:nil]);
-                         }
-                       }];
+  long errorCode = noError;
+  __block NSError *err = nil;
```

Comment:
I think it would be clearer to declare `err` without `__block`, and instead copy the value inside the block.

Reply:
Done.